addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCBridgeHandler.java

```diff
@@ -0,0 +1,471 @@
+import org.openhab.binding.plclogo.config.PLCLogoBridgeConfiguration;
+import org.openhab.binding.plclogo.internal.PLCLogoClient;
+import org.openhab.binding.plclogo.internal.PLCLogoDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import Moka7.S7;
+import Moka7.S7Client;
+
+/**
+ * The {@link PLCBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+public class PLCBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PLCBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_DEVICE);
+
+    // S7 client this bridge belongs to
+    private volatile PLCLogoClient client = null;
+    private Set<PLCBlockHandler> handlers = new HashSet<PLCBlockHandler>();
+    private PLCLogoBridgeConfiguration config = getConfigAs(PLCLogoBridgeConfiguration.class);
+
+    private Calendar rtc = Calendar.getInstance();
+    private ScheduledFuture<?> rtcJob = null;
+    private final Runnable rtcReader = new Runnable() {
+        // Buffer for diagnostic data
+        private final byte[] data = { 0, 0, 0, 0, 0, 0, 0 };
+
+        @Override
+        public void run() {
+            try {
+                if (client != null) {
+                    int result = client.ReadDBArea(1, LOGO_STATE.intValue(), data.length, S7Client.S7WLByte, data);
+                    if (result == 0) {
+                        Calendar calendar = PLCLogoDataType.getRtcAt(data, 1);
+                        if (LOGO_0BA7.equalsIgnoreCase(getLogoFamily())) {
+                            calendar = Calendar.getInstance();
+                        }
+                        synchronized (rtc) {
+                            rtc.setTimeZone(calendar.getTimeZone());
+                            rtc.setTimeInMillis(calendar.getTimeInMillis());
+                        }
+                        final Channel channel = thing.getChannel(RTC_CHANNEL_ID);
+                        updateState(channel.getUID(), new DateTimeType(rtc));
+
+                        if (logger.isTraceEnabled()) {
+                            final String raw = Arrays.toString(data);
+                            final String type = channel.getAcceptedItemType();
+                            logger.trace("Channel {} accepting {} received {}.", channel.getUID(), type, raw);
+                        }
+                    } else {
+                        logger.error("Can not read diagnostics from LOGO!: {}.", S7Client.ErrorText(result));
+                    }
+                } else {
+                    logger.error("LOGO! client {} is invalid.", client);
+                }
+            } catch (Exception exception) {
+                logger.error("RTC thread got exception: {}.", exception.getMessage());
+            } catch (Error error) {
```

Comment:
You are right, i can't do anything in this case, except logging: There were situations, that scheduler simply stop running the thread. And there was no information in log file or somewhere else. We just wondered, why binding stopped to receive values updated by LOGO! program.